rust/agents/relayer/src/msg/metadata_builder.rs

```diff
@@ -42,40 +43,56 @@ impl MetadataBuilder {
         message: &HyperlaneMessage,
         mailbox: CachingMailbox,
     ) -> eyre::Result<Option<Vec<u8>>> {
+        const CTX: &str = "When fetching metadata";
+
         // The Mailbox's `recipientIsm` function will revert if
         // the recipient is not a contract. This can pose issues with
         // our use of the RetryingProvider, which will continuously retry
         // the eth_call to the `recipientIsm` function.
         // As a workaround, we avoid the call entirely if the recipient is
         // not a contract.
         let provider = mailbox.provider();
-        if !provider.is_contract(&message.recipient).await? {
-            info!(
+        if !provider
```

Comment:
I am really tempted to create a proc macro to do this. It would be pretty easy to have something like
```rs
#[context("When doing the thing")]
fn do_thing() -> Result<()> {
  ...
}
```

which just wraps the resulting Result with a `.context` call.